### PR TITLE
fix: normaliseSessionScope and add tests

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change

Fixes the `normaliseSessionScopeOrThrowError` function and adds tests for it.

## Related issues

-  https://github.com/supertokens/supertokens-python/pull/496
-  https://github.com/supertokens/supertokens-node/pull/825

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/config_utils.go` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If access token structure has changed
    -   Modified test in `session/accessTokenVersions_test.go` to account for any new claims that are optional or omitted by the core

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [0.18.1] - 2024-05-01
```
Comment (rishabhpoddar) [inline CHANGELOG.md:10]: this should be on top of the other duplicate cookie pr (as you had said)

File: recipe/session/session_utils_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
```
Comment (rishabhpoddar) [inline recipe/session/session_utils_test.go:2]: name of this file should be sessionUtils_tests.go (see how other test files are named)

File: recipe/session/session_utils_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+package session
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNormaliseSessionScope(t *testing.T) {
```
Comment (rishabhpoddar) [inline recipe/session/session_utils_test.go:25]: add extra test cases from node pr

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by anku255** [comment]
Closing this. All the changes in this PR are moved to https://github.com/supertokens/supertokens-golang/pull/409

**Pull request closed without merging**